### DISCUSSION TITLE
Is there any way to extend the core functionalities without modifying the core files?

### DISCUSSION THREAD

**Discussion by sachin-sanchania** [venturedrake/laravel-crm#66] [Q&A]
Hello,

I need to create a custom "Leads Create" view with specific changes while ensuring that I do not modify the core files directly. Is there a recommended way to extend or override the core functionalities to achieve this customization?

**Comment by andrewdrake** [comment]
I need to write a tutorial about this :)

There are a few approaches to this and depends on what you want to do exactly. Do you need to modify the controller as well as the view file?

**Reply by sachin-sanchania** [reply]
Thank you for the quick response.

I would like to make changes to the "Leads" section, specifically in the Product section, as my business provides services rather than products. This will likely require modifications to both the controller and possibly migrations as well. 
Could you please provide a brief overview of how to properly override core functionality without modifying core files directly? Additionally, I would appreciate it if you could have any relevant documentation or approximate time for tutorial. Thank you.
